lib/evaluate/shape.h

```diff
@@ -22,6 +22,7 @@
 #include "tools.h"
 #include "type.h"
 #include "../common/indirection.h"
+#include "../parser/message.h"
```

Comment:
I don't know if it matters, but you could use a forward declaration of ContextualMessages and move the include to shape.cc.

Reply:
Sure.